### ISSUE TITLE
OLM OperatorConfig fails to install

### ISSUE THREAD

**Issue by trevor-viljoen** [external-secrets/external-secrets#563]
This issue would be better created in the `external-secrets-helm-operator `repo, but Issues don't seem to be enabled over there. I am trying to install External Secrets via the Operator Hub in OpenShift 4.8.24. I am able to create a subscription and install the operator. However, when trying to create an `OperatorConfig` using the sample provided in `external-secrets/external-secrets-helm-operator`, it fails with this error.

```
failed to install release: rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition "clustersecretstores.external-secrets.io" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "sample"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "external-secrets-operator"
```
This `OperatorConfig` fails with the same error as above.
```
apiVersion: operator.external-secrets.io/v1alpha1
kind: OperatorConfig
metadata:
  annotations:
    meta.helm.sh/release-name: external-secrets
    meta.helm.sh/release-namespace: external-secrets-operator
  name: external-secrets
  namespace: external-secrets-operator
  labels:
    app.kubernetes.io/managed-by: Helm
spec:
  nodeSelector: {}
  imagePullSecrets: []
  podLabels:
    app.kubernetes.io/managed-by: Helm
  resources: {}
  leaderElect: false
  fullnameOverride: ''
  affinity: {}
  prometheus:
    enabled: true
    service:
      port: 8080
  podSecurityContext: {}
  scopedNamespace: ''
  extraArgs: {}
  securityContext: {}
  rbac:
    create: true
  replicaCount: 1
  nameOverride: ''
  serviceAccount:
    annotations: {}
    create: true
    name: ''
  installCRDs: false
  image:
    pullPolicy: IfNotPresent
    repository: ghcr.io/external-secrets/external-secrets
    tag: ''
  tolerations: []
  extraEnv: []
  priorityClassName: ''
  podAnnotations:
    meta.helm.sh/release-name: external-secrets
    meta.helm.sh/release-namespace: external-secrets-operator
```

**Comment by trevor-viljoen** [comment]
Additionally, The events for OperatorConfig show this 18 times in the last 16 minutes "generated from operatorconfig-controller":
```
Chart value "installCRDs" overridden to "false" by operator's watches.yaml
```

**Comment by trevor-viljoen** [comment]
Here are the logs from the controller: 
```
{"level":"error","ts":1641370115.5377018,"logger":"helm.controller","msg":"Release failed","namespace":"external-secrets-operator","name":"external-secrets","apiVersion":"operator.external-secrets.io/v1alpha1","kind":"OperatorConfig","release":"external-secrets","error":"failed to install release: rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition \"clustersecretstores.external-secrets.io\" in namespace \"\" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key \"app.kubernetes.io/managed-by\": must be set to \"Helm\"; annotation validation error: missing key \"meta.helm.sh/release-name\": must be set to \"external-secrets\"; annotation validation error: missing key \"meta.helm.sh/release-namespace\": must be set to \"external-secrets-operator\"","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/co...
{"level":"error","ts":1641370115.545276,"logger":"controller.operatorconfig-controller","msg":"Reconciler error","name":"external-secrets","namespace":"external-secrets-operator","error":"failed to install release: rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition \"clustersecretstores.external-secrets.io\" in namespace \"\" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key \"app.kubernetes.io/managed-by\": must be set to \"Helm\"; annotation validation error: missing key \"meta.helm.sh/release-name\": must be set to \"external-secrets\"; annotation validation error: missing key \"meta.helm.sh/release-namespace\": must be set to \"external-secrets-operator\"","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.10.0/pkg/internal/controller/controller.go:227"}
```

I tried adding the label and annotations to an existing `ClusterSecretStore`, but it had no effect. I also tried deleting it and also deleting and reinstalling the operator, but received the same error indicating that the controller still thinks the resource exists. Is this cached somewhere?

**Comment by wseaton** [comment]
I'm running into the same exact issue, @trevor-viljoen did you have any luck installing via the Helm chart directly?

**Comment by trevor-viljoen** [comment]
@wseaton I had previously installed via Helm in another cluster with no issues. We've been holding up a production deployment, waiting for the OLM operator. So I was excited to finally get that work off my plate and marked done. I will continue debugging this today.

**Comment by wseaton** [comment]
@trevor-viljoen I switched to using a bone-stock Helm chart via ArgoCD now and that is working fine for me (now stuck debugging the providers), ideally I'd like to install via OLM as that is the more 'production-y' way of doing things in my shop as well :slightly_smiling_face:

**Comment by trevor-viljoen** [comment]
@wseaton I am deploying this via GitOps/ArgoCD and I thought initially that it was failing because I had already created a `ClusterSecretStore` as that is what one of the errors seems to be complaining about (not having proper annotations and labels). But even after adding those annotations and labels, or deleting it, the error persists.

**Comment by trevor-viljoen** [comment]
As best I can tell, the `installCRDs: false` in the `watches.yaml` is not working as intended. If you remove the CRDs that were added by OLM and let the helm operator re-add them, the ReplicaSet gets created and everything seems happy. As @slopezz mentioned here: https://github.com/external-secrets/external-secrets/issues/493#issuecomment-983542052_  OLM should manage the CRDs. You should also be able to add to the CRDs themselves the labels and annotations that the error mentions.

**Comment by trevor-viljoen** [comment]
Running this prior to creating the `OperatorConfig` gets around the problem. It's more of a hack than a solution.

@wseaton 
```shell
release_name="cluster"
release_namespace="external-secrets"

for crd in clustersecretstores externalsecrets secretstores ; do
  kubectl label crd ${crd}.external-secrets.io app.kubernetes.io/managed-by=Helm
  kubectl annotate crd ${crd}.external-secrets.io meta.helm.sh/release-name=${release-name} meta.helm.sh/release-namespace=${release-namespace}
done
```

**Comment by knelasevero** [comment]
I thought about enabling issues on that repo, but I am now thinking I prefer all issues here on the main repo and organize them by tags. Will bring this up at the next community meeting.

@moolen was the one pushing this forward, and I think he is still on vacation and unavailable, but I can try to have a look. I am a bit unfamiliar with OLM and with the work done with the es-helm-operator. I see we suspect the culprit is the `installCRDs: false` in the `watches.yaml` file of that repo. Does anybody have a clue why that is not preventing the operator from fighting with OLM on who manages the crds?

I will try to debug this a bit this week.

**Comment by knelasevero** [comment]
I can confirm the issue.

I created this PR on the other repo:

https://github.com/external-secrets/external-secrets-helm-operator/pull/1

Will merge soon and after that, I will create the PR on the community_operators repo to point to the new release.

Probably will also release ESO 0.3.11 already as well.

**Comment by knelasevero** [comment]
https://github.com/k8s-operatorhub/community-operators/pull/550

I will ping here when it gets published to https://operatorhub.io/

Then we can check if it works for you guys

**Comment by knelasevero** [comment]
PR was merged, it should show up in https://operatorhub.io/ in about 10-30 min

**Comment by knelasevero** [comment]
I just tried this with OLM for 0.3.11. Works like a charm. Closing the issue. Feel free to re-open in case of any problems!

**Comment by trevor-viljoen** [comment]
@knelasevero Thank you for the quick turn around on this.

**Comment by tviljoen-faa** [comment]
@knelasevero There probably needs to be a corresponding PR here as well: https://github.com/redhat-openshift-ecosystem/community-operators-prod/ as done here: https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/570

**Comment by knelasevero** [comment]
Oh, I missed that one. Gonna do it tomorrow

**Comment by knelasevero** [comment]
https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/579

Merged :)

**Comment by trevor-viljoen** [comment]
The automatic upgrade failed and I did have to uninstall the operator to allow it to re-install the CRDs after having made the annotation and label changes mentioned here https://github.com/external-secrets/external-secrets/issues/563#issuecomment-1006142592. These effectively gave control of the CRDs to Helm and were removed as part of the upgrade to 0.3.11. That said, after re-installing the operator everything looks good from what I can see.
